test/jdk/java/foreign/NativeTestHelper.java

```diff
@@ -93,44 +93,63 @@ public static boolean isPointer(MemoryLayout layout) {
         return layout instanceof ValueLayout valueLayout && valueLayout.carrier() == MemorySegment.class;
     }
 
+    public static final Linker LINKER = Linker.nativeLinker();
+
     // the constants below are useful aliases for C types. The type/carrier association is only valid for 64-bit platforms.
 
     /**
      * The layout for the {@code bool} C type
      */
-    public static final ValueLayout.OfBoolean C_BOOL = ValueLayout.JAVA_BOOLEAN;
+    public static final ValueLayout.OfBoolean C_BOOL = (ValueLayout.OfBoolean) LINKER.canonicalLayouts().get("bool");
     /**
      * The layout for the {@code char} C type
      */
-    public static final ValueLayout.OfByte C_CHAR = ValueLayout.JAVA_BYTE;
+    public static final ValueLayout.OfByte C_CHAR = (ValueLayout.OfByte) LINKER.canonicalLayouts().get("char");
     /**
      * The layout for the {@code short} C type
      */
-    public static final ValueLayout.OfShort C_SHORT = ValueLayout.JAVA_SHORT;
+    public static final ValueLayout.OfShort C_SHORT = (ValueLayout.OfShort) LINKER.canonicalLayouts().get("short");
     /**
      * The layout for the {@code int} C type
      */
-    public static final ValueLayout.OfInt C_INT = ValueLayout.JAVA_INT;
+    public static final ValueLayout.OfInt C_INT = (ValueLayout.OfInt) LINKER.canonicalLayouts().get("int");
 
     /**
      * The layout for the {@code long long} C type.
      */
-    public static final ValueLayout.OfLong C_LONG_LONG = ValueLayout.JAVA_LONG;
+    public static final ValueLayout.OfLong C_LONG_LONG = (ValueLayout.OfLong) LINKER.canonicalLayouts().get("long long");
     /**
      * The layout for the {@code float} C type
      */
-    public static final ValueLayout.OfFloat C_FLOAT = ValueLayout.JAVA_FLOAT;
+    public static final ValueLayout.OfFloat C_FLOAT = (ValueLayout.OfFloat) LINKER.canonicalLayouts().get("float");
     /**
      * The layout for the {@code double} C type
      */
-    public static final ValueLayout.OfDouble C_DOUBLE = ValueLayout.JAVA_DOUBLE.withByteAlignment(isAixOS ? 4 : Double.BYTES);
+    public static final ValueLayout.OfDouble C_DOUBLE = isAixOS ? ValueLayout.JAVA_DOUBLE.withByteAlignment(4) : (ValueLayout.OfDouble) LINKER.canonicalLayouts().get("double");
```

Comment:
This file can be identical to upstream with my suggested change to `AixPPC64Linker`.

Reply:
Restored upstream version.